controllers/reconciler.go

```diff
@@ -411,8 +411,9 @@ func deploymentForTunnel(r GenericTunnelReconciler) *appsv1.Deployment {
 	replicas := r.GetTunnel().GetSpec().Size
 	nodeSelector := nodeSelectorsForTunnel(r.GetTunnel())
 	tolerations := r.GetTunnel().GetSpec().Tolerations
+	protocol := r.GetTunnel().GetSpec().Protocol
 
-	args := []string{"tunnel", "--config", "/etc/cloudflared/config/config.yaml", "--metrics", "0.0.0.0:2000", "run"}
+	args := []string{"tunnel", "--protocol", protocol, "--config", "/etc/cloudflared/config/config.yaml", "--metrics", "0.0.0.0:2000", "run"}
```

Comment:
Could we somehow validate the value of protocol, either at the crd level or atleast in the app? This would cause the pod to fail when created.